gto-support-androidx-fragment/src/main/kotlin/org/ccci/gto/android/common/androidx/fragment/app/FragmentManager.kt

```diff
@@ -0,0 +1,20 @@
+package org.ccci.gto.android.common.androidx.fragment.app
+
+import androidx.fragment.app.FragmentManager
+
+internal fun FragmentManager.backStackEntriesIterator() = object : ListIterator<FragmentManager.BackStackEntry> {
+    private val manager = this@backStackEntriesIterator
+    private var i = 0
+
+    override fun hasNext() = i < manager.backStackEntryCount
+    override fun next() = manager.getBackStackEntryAt(i++)
+    override fun hasPrevious() = i > 0
+    override fun previous() = manager.getBackStackEntryAt(--i)
+    override fun nextIndex() = i
```

Comment:
Maybe I'm misunderstanding but wouldn't the next index be ```i + 1``` similar to how ```previousIndex``` is ```i - 1```? Doesn't ```i``` represent the current index?

Reply:
in this logic `i` represents the index of the next item that will be returned by `next()`.

the postfix increment operator in `next()` might be throwing you off here. `i++` will return the current value of i, and then increment i to the next value.

This is different from the prefix decrement operator on `previous()` that will decrement i, then return the new value of i.

Reply:
if needed I can dig into this more with you, just ping me on slack

Reply:
Aaaah okay, that makes sense. Yeah, that's exactly what was throwing me off, thanks for explaining!